### ISSUE TITLE
LRScheduler is stepped with loss parameter instead of epoch

### ISSUE THREAD

**Issue by leonschindler97** [MKHoffmann/icpReconstructor#3]
The LRScheduler used in TorchCurveEstimator.fit() is stepped with the loss value instead of the current epoch. However, passing the epoch parameter to the LRScheduler.step() is deprecated and should not be used.

**Proposed fix**
Move the scheduler.step() call at the end of the outer epoch loop.
